app/models/exporters/tableau.rb

```diff
@@ -4,5 +4,185 @@ module Exporters::Tableau
+        hh_config:                        she_t[:presented_as_individual],
+        prov_id:                          she_t[:project_name],
+        _prov_id:                         she_t[:project_id],
+        prog_type:                        she_t[model.project_type_column],
+        prov_jurisdiction:                site_t[:City],
+        entry_exit_entry_date:            she_t[:first_date_in_program],
+        entry_exit_exit_date:             she_t[:last_date_in_program],
+        client_dob:                       c_t[:DOB],
+        client_age_at_entry:              she_t[:age],
+        client_6orunder:                  nil,
+        client_7to17:                     nil,
+        client_18to24:                    nil,
+        veteran_status:                   c_t[:VeteranStatus],
+        hispanic_latino:                  c_t[:Ethnicity],
+        **c_t.engine.race_fields.map{ |f| [ "primary_race_#{f}".to_sym, c_t[f.to_sym] ] }.to_h, # primary race logic is funky
+        disabling_condition:              d_t[:DisabilitiesID],
+        any_income_30days:                nil,
+        county_homeless:                  null, # ???
+        res_prior_to_entry:               e_t[:ResidencePrior],
+        length_of_stay_prev_place:        e_t[:ResidencePriorLengthOfStay],
+        approx_date_homelessness_started: e_t[:DateToStreetESSH],
+        times_on_street:                  e_t[:TimesHomelessPastThreeYears],
+        total_months_homeless_on_street:  e_t[:MonthsHomelessPastThreeYears],
+        destination:                      she_t[:destination],
+        destination_other:                ex_t[:OtherDestination],
+        service_uid:                      shs_t[:id],
+        service_inactive:                 null, # ???
+        service_code_desc:                shs_t[:service_type],
+        service_start_date:               shs_t[:date],
+        # after this point in the sample data everything is NULL
+        entry_exit_uid:                   null,
+        days_to_return:                   null,
+        entry_exit_uid:                   null, # REPEAT
+        days_last3years:                  null,
+        instances_last3years:             null,
+        entry_exit_uid:                   null, # REPEAT
+        rrh_time_in_shelter:              null,
+      }
+
+      scope = model.residential.entry.
+        open_between( start_date: start_date, end_date: end_date ).
+        joins( :client, :service_history_services ).
+        joins( project: :sites, enrollment: :exit ).
+        includes( client: :source_disabilities ).
+        references( client: :source_disabilities ).
+        # for aesthetics
+        order( she_t[:client_id].asc ).
+        order( e_t[:id].asc ).
+        order( she_t[:first_date_in_program].desc ).
+        order( she_t[:last_date_in_program].desc ).
+        order( shs_t[:id].asc )
+
+      scope = case coc_code
+      when Array
+        scope.where( pc_t[:CoCCode].in coc_code )
```

Comment:
This needs to take into account the CoCCode override see `GrdaWarehouse::Hud::ProjectCoc.in_coc`

Reply:
Dang. It seems like you could coalesce `hud_coc_code` and `CoCCode` there, but maybe that wouldn't be as dbms-agnostic. Anyway, I'll fix it.